tests/integration/security/authorization_test.go

```diff
@@ -228,78 +222,78 @@ func TestAuthorization_WorkloadSelector(t *testing.T) {
-							NamePrefix: namePrefix,
-							Request: connection.Checker{
-								From: a[0],
-								Options: echo.CallOptions{
-									Target:   target[0],
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Path:     path,
-									Count:    callCount,
+
+					applyPolicy("testdata/authz/v1beta1-workload-ns1.yaml.tmpl", ns1)
+					applyPolicy("testdata/authz/v1beta1-workload-ns2.yaml.tmpl", ns2)
+					applyPolicy("testdata/authz/v1beta1-workload-ns-root.yaml.tmpl", rootns)
+					t.NewSubTest(fmt.Sprintf("From %s", srcCluster.StableName())).Run(func(t framework.TestContext) {
+						callCount := 1
+						if t.Clusters().IsMulticluster() {
+							// so we can validate all clusters are hit
+							callCount = util.CallsPerCluster * len(t.Clusters())
+						}
+						newTestCase := func(namePrefix string, target echo.Instances, path string, expectAllowed bool) rbacUtil.TestCase {
+							return rbacUtil.TestCase{
+								NamePrefix: namePrefix,
+								Request: connection.Checker{
+									From: a[0],
+									Options: echo.CallOptions{
+										Target:   target[0],
+										PortName: "http",
+										Scheme:   scheme.HTTP,
+										Path:     path,
+										Count:    callCount,
+									},
+									DestClusters: target.Clusters(),
 								},
-								DestClusters: target.Clusters(),
-							},
-							ExpectAllowed: expectAllowed,
+								ExpectAllowed: expectAllowed,
+							}
 						}
-					}
-					cases := []rbacUtil.TestCase{
-						newTestCase("[bInNS1]", bInNS1, "/policy-ns1-b", true),
-						newTestCase("[bInNS1]", bInNS1, "/policy-ns1-c", false),
-						newTestCase("[bInNS1]", bInNS1, "/policy-ns1-x", false),
-						newTestCase("[bInNS1]", bInNS1, "/policy-ns1-all", true),
-						newTestCase("[bInNS1]", bInNS1, "/policy-ns2-c", false),
-						newTestCase("[bInNS1]", bInNS1, "/policy-ns2-all", false),
-						newTestCase("[bInNS1]", bInNS1, "/policy-ns-root-c", false),
-
-						newTestCase("[cInNS1]", cInNS1, "/policy-ns1-b", false),
-						newTestCase("[cInNS1]", cInNS1, "/policy-ns1-c", true),
-						newTestCase("[cInNS1]", cInNS1, "/policy-ns1-x", false),
-						newTestCase("[cInNS1]", cInNS1, "/policy-ns1-all", true),
-						newTestCase("[cInNS1]", cInNS1, "/policy-ns2-c", false),
-						newTestCase("[cInNS1]", cInNS1, "/policy-ns2-all", false),
-						newTestCase("[cInNS1]", cInNS1, "/policy-ns-root-c", true),
-
-						newTestCase("[cInNS2]", cInNS2, "/policy-ns1-b", false),
-						newTestCase("[cInNS2]", cInNS2, "/policy-ns1-c", false),
-						newTestCase("[cInNS2]", cInNS2, "/policy-ns1-x", false),
-						newTestCase("[cInNS2]", cInNS2, "/policy-ns1-all", false),
-						newTestCase("[cInNS2]", cInNS2, "/policy-ns2-c", true),
-						newTestCase("[cInNS2]", cInNS2, "/policy-ns2-all", true),
-						newTestCase("[cInNS2]", cInNS2, "/policy-ns-root-c", true),
-					}
-					rbacUtil.RunRBACTest(t, cases)
-				})
+
+						newTestCases := func(target echo.Instances) []rbacUtil.TestCase {
```

Comment:
it's really hard to track what is being tested and its expected output in this form, can we change back to the old table-driven definition?

Reply:
Agree on the bad readability. I have rewritten them in a table-driven format.